src/meshpy/geometric_search/__init__.py

```diff
@@ -19,4 +19,28 @@
 # LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 # THE SOFTWARE.
-"""This module defines geometric search functionality."""
+"""This module defines geometric search functionality.
+
+This module contains functionality to find points close to each other in
+a point cloud. Currently, three different implementations for the actual
+search algorithm are available (depending on your installation/setup):
+
+- `brute_force_cython`: A brute force algorithm implemented in Cython,
+  scales with $\mathcal{O}(n^2)$, but for a small number of points
+  ($n<200$) this is the fastest algorithm since compared to the others
+  it does not have any setup costs.
+
+- `kd_tree_scipy`: Uses a BVH implementation provided by scipy. This
+  scales with $\mathcal{O}(n\ \log{n})$.
+
+- `boundary_volume_hierarchy_arborx` : Uses a BVH implementation
+  provided by ArborX. This also scales with $\mathcal{O}(n\ \log{n})$
+  but due to a more optimised implementation is a few times faster
+  than the scipy implementation.
+
+The algorithm automatically chooses the fastest (available)
```

Comment:
Sorry I forgot to add this in the initial review

```suggestion
The `find_close_points` function automatically chooses the fastest (available)
```